crates/turbo-tasks/src/manager.rs

```diff
@@ -1487,23 +1488,24 @@ impl CurrentCellRef {
         tt.update_own_task_cell(
             self.current_task,
             self.index,
-            CellContent(Some(SharedReference::new(
-                Some(self.index.type_id),
-                triomphe::Arc::new(new_content),
-            ))),
+            CellContent(Some(SharedReference::new(triomphe::Arc::new(new_content)))),
         )
     }
 
-    pub fn update_shared_reference(&self, shared_ref: SharedReference) {
+    pub fn update_shared_reference(&self, shared_ref: TypedSharedReference) {
```

Comment:
The type info for `shared_ref` isn't used here? Maybe just pass a `SharedReference` or validate the type info with the cell type.